File: organization-name-standardization/python-examples/standardize-file.go
package main

import (
	"bufio"
	"encoding/csv"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"net/url"
	"os"
)

// Replace with your key: https://www.interzoid.com/manage-api-account
const API_KEY = "YOUR_API_KEY_HERE"

const inputFileName = "sample-input-file.txt"
const outputFileName = "output.csv"

func main() {

	// Open the input file
	inFile, err := os.Open(inputFileName)
	if err != nil {
		fmt.Println("Error opening input file:", err)
		return
	}
	defer inFile.Close()

	// Create output CSV file
	outFile, err := os.Create(outputFileName)
	if err != nil {
		fmt.Println("Error creating output file:", err)
		return
	}
	defer outFile.Close()

	csvWriter := csv.NewWriter(outFile)
	defer csvWriter.Flush()

	// JSON mapping struct
	type Response struct {
		Standard string `json:"Standard"`
		Code     string `json:"Code"`
		Credits  string `json:"Credits"`
	}

	scanner := bufio.NewScanner(inFile)
	lineNumber := 0

	for scanner.Scan() {
		lineNumber++
		originalValue := scanner.Text()

		// Allow blank lines to be skipped
		if len(originalValue) == 0 {
			continue
		}

		// Encode organization name
		orgParam := url.QueryEscape(originalValue)

		// Build the request URL
		apiURL := "https://api.interzoid.com/getorgstandard?license=" + API_KEY +
			"&org=" + orgParam

		// Issue request
		resp, err := http.Get(apiURL)
		if err != nil {
			fmt.Printf("API call error for line %d (%q): %v\n", lineNumber, originalValue, err)
			_ = csvWriter.Write([]string{originalValue, ""})
			continue
		}

		body, err := ioutil.ReadAll(resp.Body)
		resp.Body.Close()
		if err != nil {
			fmt.Printf("Error reading response for %q: %v\n", originalValue, err)
			_ = csvWriter.Write([]string{originalValue, ""})
			continue
		}

		var result Response
		if err := json.Unmarshal(body, &result); err != nil {
			fmt.Printf("JSON parse error for %q: %v\n", originalValue, err)
			_ = csvWriter.Write([]string{originalValue, ""})
			continue
		}

		// Write row to CSV
		err = csvWriter.Write([]string{originalValue, result.Standard})
		if err != nil {
			fmt.Printf("CSV write error for %q: %v\n", originalValue, err)
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading input file:", err)
		return
	}

	fmt.Println("Done. Results written to", outputFileName)
}

File: organization-name-standardization/node-examples/standardize-file.js
// Simple Node.js example that:
// - Reads organization names from a text file (one per line)
// - Calls Interzoid's Organization Standardization API
// - Writes original value + standardized value to a CSV file
//
// Requirements:
// - Node.js 18+ (for built-in fetch)
// - No third-party libraries

const fs = require("fs");
const readline = require("readline");

// 1. Replace this with your API key from https://www.interzoid.com/manage-api-account
const API_KEY = "YOUR_API_KEY_HERE";

// 2. Input and output file names
const inputFileName = "sample-input-file.txt";
const outputFileName = "output.csv";

// Helper to safely escape values for CSV:
// - Wraps in quotes if needed
// - Doubles any existing quotes
function csvEscape(value) {
  const str = String(value);
  if (/[",\r\n]/.test(str)) {
    return `"${str.replace(/"/g, '""')}"`;
  }
  return str;
}

async function main() {
  // Open the input file as a readable stream
  let inputStream;
  try {
    inputStream = fs.createReadStream(inputFileName, { encoding: "utf8" });
  } catch (err) {
    console.error("Error opening input file:", err.message);
    return;
  }

  // Create the output CSV file as a writable stream
  let outputStream;
  try {
    outputStream = fs.createWriteStream(outputFileName, { encoding: "utf8" });
  } catch (err) {
    console.error("Error creating output file:", err.message);
    return;
  }

  // Use readline to process the input file line by line
  const rl = readline.createInterface({
    input: inputStream,
    crlfDelay: Infinity,
  });

  let lineNumber = 0;

  for await (const line of rl) {
    lineNumber++;
    const originalValue = line;

    // Skip blank lines if desired
    if (originalValue.length === 0) {
      continue;
    }

    // Build the API URL. URLSearchParams will safely encode the org name.
    const endpoint = "https://api.interzoid.com/getorgstandard";
    const params = new URLSearchParams({
      license: API_KEY,
      org: originalValue,
    });

    const apiURL = `${endpoint}?${params.toString()}`;

    try {
      // Call the API
      const response = await fetch(apiURL);

      if (!response.ok) {
        console.error(
          `API call error for line ${lineNumber} (${JSON.stringify(
            originalValue
          )}): HTTP ${response.status} ${response.statusText}`
        );
        // Still write a row with an empty standardized value
        outputStream.write(
          `${csvEscape(originalValue)},${csvEscape("")}\n`
        );
        continue;
      }

      // Parse the JSON response
      const data = await response.json();

      // The response contains a "Standard" field with the standardized organization name
      const standard = data.Standard || "";

      // Write row to CSV: originalValue, standardizedValue
      outputStream.write(
        `${csvEscape(originalValue)},${csvEscape(standard)}\n`
      );
    } catch (err) {
      console.error(
        `Error processing line ${lineNumber} (${JSON.stringify(
          originalValue
        )}):`,
        err.message
      );
      // On error, still write a row with empty standardized value
      outputStream.write(
        `${csvEscape(originalValue)},${csvEscape("")}\n`
      );
    }
  }

  // Close the output file when finished
  outputStream.end(() => {
    console.log("Done. Results written to", outputFileName);
  });
}

// Run the script
main().catch((err) => {
  console.error("Unexpected error:", err.message);
});
